File: Robot Mesh Studio/C++/Bump Sensor.cpp
//LINK TO SOURCE: https://www.robotmesh.com/studio/5be40c75c8f17a1f57963006
// VEX V5 C++ Project
#include "vex.h"

//#region config_globals
vex::brain      Brain;
vex::motor      motor_right(vex::PORT15, vex::gearSetting::ratio18_1, true);
vex::motor      motor_left(vex::PORT16, vex::gearSetting::ratio18_1, false);
vex::bumper     bumper_back(Brain.ThreeWirePort.A);
vex::drivetrain dt(motor_left, motor_right, 319.1764, 292.1, vex::distanceUnits::mm);
//#endregion config_globals

using namespace vex;

int main(void) {
    dt.setVelocity(20,velocityUnits::pct); //take it slow
    while(true) {
        //creep backwards...
        dt.drive(directionType::rev);
        while(!bumper_back.pressing()){} //wait for bumper to be pressed
        dt.drive(directionType::fwd);
        sleep(2); //drive away for two seconds
        dt.turnFor(turnType::left,77,rotationUnits::deg); //turn to avoid obstacle
        //ready for another try
    }
}

File: Robot Mesh Studio/Offical Vex examples/Python/Bump Sensor.py
#LINK TO SOURCE: https://www.robotmesh.com/studio/5be40c6ec8f17a1f5795f5b1
# VEX V5 Python Project
import sys
import vex
from vex import *

#region config
brain       = vex.Brain();
motor_right = vex.Motor(vex.Ports.PORT15, vex.GearSetting.RATIO18_1, False)
motor_left  = vex.Motor(vex.Ports.PORT16, vex.GearSetting.RATIO18_1, False)
bumper_a    = vex.Bumper(brain.three_wire_port.a)
#endregion config

# Carefully now...
dt.set_velocity(20, vex.VelocityUnits.PCT)
while True:
  # Creep back...
  dt.drive(vex.DirectionType.REV)
  sys.wait_for(bumper_a.pressing)
  # Ran into something, move away from it
  dt.drive(vex.DirectionType.FWD)
  sys.sleep(2)
  # Turn a bit to avoid it...
  dt.turn_for(vex.TurnType.LEFT, 77, vex.RotationUnits.DEG)
  # Ready to try again!